Common/src/interpolation_structure.cpp

```diff
@@ -1352,7 +1361,1770 @@ void CMirror::Set_TransferCoeff(CConfig **config) {
+
+    iElementNode++;
+
+    for (iDim = 0; iDim < nDim; iDim++) 
+      element[ iElementNode ][iDim] = ( element[0][iDim] + coord[ OuterNodes[ CurrentNode ] * nDim + iDim] + coord[ OuterNodes[ NextNode ] * nDim + iDim] )/3;
+
+    iElementNode++;
+
+    if( OuterNodesNeighbour[ NextNode ][0] == CurrentNode){
+      CurrentNode = NextNode; 
+      NextNode = OuterNodesNeighbour[ NextNode ][1];  
+    }
+    else{
+      CurrentNode = NextNode; 
+      NextNode = OuterNodesNeighbour[ NextNode ][0];  
+    }
+
+    if (CurrentNode == StartIndex)
+      break;
+    }
+
+    if( CurrentNode == StartIndex ){ // This is a closed element, so add again element 1 to the end of the structure, useful later
+
+    for (iDim = 0; iDim < nDim; iDim++)
+      element[ iElementNode ][iDim] = element[1][iDim];
+    iElementNode++;
+  }
+  else{
+    for (iDim = 0; iDim < nDim; iDim++)
+    element[ iElementNode ][iDim] = ( element[0][iDim] + coord[ OuterNodes[ CurrentNode ] * nDim + iDim] )/2;
+    iElementNode++;
+  }
+
+  for ( iNode = 0; iNode < nOuterNodes; iNode++ )
+    delete [] OuterNodesNeighbour[ iNode ];
+  delete [] OuterNodesNeighbour;
+
+  return iElementNode;
+}
+
+su2double CSlidingmesh::ComputeLineIntersectionLength(su2double* A1, su2double* A2, su2double* B1, su2double* B2, su2double* Direction){
+    
+  /*--- Given 2 segments, each defined by 2 points, it projects them along a given direction and it computes the length of the segment resulting from their intersection ---*/
+  /*--- The algorithm works for both 2D and 3D problems ---*/
+
+  unsigned short iDim;
+  unsigned short nDim = donor_geometry->GetnDim();
+
+  su2double dotA2, dotB1, dotB2;
+
+  dotA2 = 0;
+  for(iDim = 0; iDim < nDim; iDim++)
+    dotA2 += ( A2[iDim] - A1[iDim] ) * Direction[iDim];
+
+  if( dotA2 >= 0 ){
+    dotB1 = 0;
+    dotB2 = 0;
+    for(iDim = 0; iDim < nDim; iDim++){
+      dotB1 += ( B1[iDim] - A1[iDim] ) * Direction[iDim];
+      dotB2 += ( B2[iDim] - A1[iDim] ) * Direction[iDim];
+    }
+  }
+  else{
+    dotA2 *= -1;
+
+    dotB1 = 0;
+    dotB2 = 0;
+    for(iDim = 0; iDim < nDim; iDim++){
+      dotB1 -= ( B1[iDim] - A1[iDim] ) * Direction[iDim];
+      dotB2 -= ( B2[iDim] - A1[iDim] ) * Direction[iDim];
+    }
+  }
+
+  if( dotB1 >= 0 && dotB1 <= dotA2 ){
+    if ( dotB2 < 0 )
+      return fabs( dotB1 );
+    if ( dotB2 > dotA2 )
+      return fabs( dotA2 - dotB1 );
+
+    return fabs( dotB1 - dotB2 );
+  }
+
+  if( dotB2 >= 0 && dotB2 <= dotA2 ){
+    if ( dotB1 < 0 )
+      return fabs(dotB2);
+    if ( dotB1 > dotA2 )
+      return fabs( dotA2 - dotB2 );
+  }
+
+  if( ( dotB1 <= 0 && dotA2 <= dotB2 ) || ( dotB2 <= 0 && dotA2 <= dotB1 ) )
+    return fabs( dotA2 );
+
+  return 0.0;
+}
+
+su2double CSlidingmesh::Compute_Triangle_Intersection(su2double* A1, su2double* A2, su2double* A3, su2double* B1, su2double* B2, su2double* B3, su2double* Direction){
```

Comment:
All of these geometry routines look nice. Do you think we could use them for other purposes? I am thinking for things like probes, cutting slices of data or geometry, etc. We have some routines for these things in other locations for SU2_GEO. Maybe we need to consolidate these routines in a general location somewhere...